Backend/server.js

```diff
@@ -120,22 +122,76 @@ app.get('/getSubmission', async (req, res) => {
 
 app.get('/logout', limiter, async (req, res) => {
   const user_id = req.query.user_id;
+  const login_time = req.query.login_time;  
+  console.log(login_time)
   
   let db = getDb();
-  db.updateOne({ "canvasUser": { $eq: user_id } }, { $set: { "lastLogout": Date.now() } })
+  db.updateOne({ "username": { $eq: user_id } }, { $set: { "lastLogin": login_time } })
   console.log("< logged out user " + user_id)
   res.status(200).json({ message: "Logged out!" });
 })
 
-app.get('/login', limiter, async (req, res) => {
-  const user_id = req.query.user_id;
+
+
+app.post('/loginUser', limiter, async (req, res) => {
+  const u = req.body.username;
+  const p = req.body.password;
   
+  if (!u || !p) {
+    return res.status(400).json({message: "Username and password required!"});
+  }
+
   let db = getDb();
-  console.log("> logged in user " + user_id)
-  let user = await db.findOne({ "canvasUser": { $eq: user_id } })
-  res.status(200).json({ user });
+  let user = await db.find({"username" : { $eq: u }})
+  user = await user.toArray();
+
+  var json;
+  var code;
+
+  if (user.length > 0) {
+    code = 401;
+    json = {message: "Incorrect password!"}
+  } else {
+    code = 404;
+    json = {message: "No users found!"}
```

Comment:
should we differentiate between these? a 404 is usually for a page that doesn't exist not something like this (it gave me some confusion while looking at browser requests)

we could just have a catch-all "incorrect username or password"